### PR TITLE
Improve doc on lowercase treatment of columns on SQL

### LINKED ISSUE #2374: Clarify in docs that Identifiers are made lower-case in SQL query
**Describe the bug**
With (I think) [the most up-to-date version](https://github.com/apache/arrow-datafusion/commit/744b2626081db95a254fc882820fc7812f95aa51) of the arrow2 branch, the idenfitiers in query seem to be made lower-case

For example if I modify the `memtable` example by renaming the `bank_acount` field to `Bank_Account` then the query will fail,

```shell
$ gd
diff --git a/datafusion-examples/examples/memtable.rs b/datafusion-examples/examples/memtable.rs
index 4c635205..16bd247d 100644
--- a/datafusion-examples/examples/memtable.rs
+++ b/datafusion-examples/examples/memtable.rs
@@ -37,7 +37,7 @@ async fn main() -> Result<()> {
     // Register the in-memory table containing the data
     ctx.register_table("users", Arc::new(mem_table))?;
 
-    let dataframe = ctx.sql("SELECT * FROM users;").await?;
+    let dataframe = ctx.sql("SELECT Bank_Account FROM users;").await?;
 
     timeout(Duration::from_secs(10), async move {
         let result = dataframe.collect().await.unwrap();
@@ -72,6 +72,6 @@ fn create_record_batch() -> Result<RecordBatch> {
 fn get_schema() -> SchemaRef {
     SchemaRef::new(Schema::new(vec![
         Field::new("id", DataType::UInt8, false),
-        Field::new("bank_account", DataType::UInt64, true),
+        Field::new("Bank_Account", DataType::UInt64, true),
     ]))
 }
$ cargo run --example memtable
    Finished dev [unoptimized + debuginfo] target(s) in 0.35s
     Running `./target/debug/examples/memtable`
Error: Plan("Invalid identifier '#bank_account' for schema fields:[users.id, users.Bank_Account], metadata:{}")
```

Meaning, if the source table has any non-lowercase characters, the queries cannot work

**Additional context**
I'm using the arrow2 branch at this revision:

https://github.com/apache/arrow-datafusion/commit/744b2626081db95a254fc882820fc7812f95aa51

### PR DESCRIPTION
**Pull request by nanicpc**
Adds specific note on lowercase treatment of columns on SQL

# Which issue does this PR close?

Closes #2374 

 # Rationale for this change
This example is the first thing newcomers see when they start with datafusion (literally the second page on the docs) so it should be clearer.

# What changes are included in this PR?
Improvements on the Documentation:
* the "Example Usage" case
* the SELECT page of SQL

### CODE REVIEW

File: docs/source/user-guide/example-usage.md [hunk 1]
```diff
@@ -40,7 +42,7 @@ async fn main() -> datafusion::error::Result<()> {
   ctx.register_csv("example", "tests/example.csv", CsvReadOptions::new()).await?;
 
   // create a plan to run a SQL query
-  let df = ctx.sql("SELECT a, MIN(b) FROM example GROUP BY a LIMIT 100").await?;
+  let df = ctx.sql("SELECT \"A\", MIN(b) FROM example GROUP BY \"A\" LIMIT 100").await?;
```
Comment (alamb) [inline docs/source/user-guide/example-usage.md:45]: I am not sure about changing this code example. When I tried this example code now using https://github.com/apache/arrow-datafusion/blob/master/datafusion/core/tests/example.csv

I get an error
```
     Running `/Users/alamb/Software/target-iox2/debug/df_test`
Error: SchemaError(FieldNotFound { qualifier: None, name: "A", valid_fields: Some(["example.a", "example.b", "example.c"]) })
```
Maybe we should add an explicit link to the example.csv file ?

File: docs/source/user-guide/example-usage.md [hunk 2]
```diff
@@ -19,6 +19,8 @@
 
 # Example Usage
 
+In this example some simple processing is performed on a csv file. Please be aware that all identifiers are made lower-case in SQL, so if your csv file has capital letters (ex: Name) you should put your column name in double quotes or the example won't work.
```
Comment (alamb) [inline docs/source/user-guide/example-usage.md:22]: This note is great

**Review by alamb — COMMENTED** [review]
Thank you @nanicpc  -- I think the notes are great. I am not sure about the change to the example code.

**Comment by alamb** [comment]
Marking as draft to signify this PR has planned changes (and make it easier to find PRs that are in need of review). Please mark "ready for review" when it is next ready or if this change was a mistake.

**Review by alamb — APPROVED** [review]
Thanks @nanicpc

**Comment by alamb** [comment]
What do you think about changing the introductory example in the user guide @andygrove ? Given it is the first thing some people see I think there is a potential reason to keep it as simple as possible (and thus maybe it would be nice to leave it uncapitalized, and add add a link about the important subtlety of capitalization to somewhere later? 

If you agree I will merge this PR and then make a follow on PR that tries to retain the simple initial example as well as the capitalized example

**Comment by andygrove** [comment]
> What do you think about changing the introductory example in the user guide @andygrove ? Given it is the first thing some people see I think there is a potential reason to keep it as simple as possible (and thus maybe it would be nice to leave it uncapitalized, and add add a link about the important subtlety of capitalization to somewhere later?
> 
> If you agree I will merge this PR and then make a follow on PR that tries to retain the simple initial example as well as the capitalized example

That sounds great. Thanks @alamb

**Comment by alamb** [comment]
Thanks again @nanicpc

**Comment by alamb** [comment]
Follow on PR in https://github.com/apache/arrow-datafusion/pull/3815

**Pull request merged**